src/foam/core/lib.js

```diff
@@ -23,11 +23,14 @@ foam = {
   isServer: globalThis.FOAM_FLAGS.node,
   core:     {},
   util:     {
-    path: function(root, path, opt_ensure) {
+    path: function(root, path, opt_ensure, strict) {
       var a = path.split('.');
       var i;
 
       for ( var i = 0 ; i < a.length ; i++ ) {
+        if ( strict && ! root[a[i]] ) {
```

Comment:
I think we should remove this and just replace line 30 with:
for ( var i = 0 ; i < a.length && root ; i++ ) {